h/groups/search.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+
+class GroupAuthFilter(object):
+    def __init__(self, request):
+        self.auth_domain = request.auth_domain
+        self.authenticated_user = request.authenticated_user
+        self.session = request.db
+        self.group_service = request.find_service(name='group')
+
+    def __call__(self, _):
+        groups = set(['__world__'])
+
+        if self.authenticated_user:
+            for group in self.authenticated_user.groups:
+                groups.add(group.pubid)
+
+        for pubid in self.group_service.world_readable_pubids():
```

Comment:
Huh? Are you sure?

```python
In [4]: readable_member = sa.and_(m.Group.readable_by == ReadableBy.members, m.Group.members.contains(user))

In [5]: print readable_member
"group".readable_by = :readable_by_1 AND "group".id = user_group_1.group_id AND :param_1 = user_group_1.user_id
```

Unless you're a member of 29,460 groups, I can't see how that query could return that many rows?